core/nunjucks.compat.js

```diff
@@ -116,13 +116,13 @@ define('nunjucks.compat', ['nunjucks'], function(nunjucks) {
             throw new Error('KeyError');
         },
         setdefault: function(key, def) {
-            if (key in this) {
-                return this[key];
-            }
-            if (def === undefined) {
-                def = null;
+            if (!(key in this)) {
+                if (def === undefined) {
```

Comment:
Ah I see no longer a problem assuming ES5 as undefined can't be overwritten any more.